File: src/main/java/com/ai/testgen/util/RequirementReader.java
package com.ai.testgen.util;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

/**
 * Utility class to read requirements from a text file
 */
public class RequirementReader {
    
    private static final String DEFAULT_INPUT_PATH = "input/requirements.txt";
    
    /**
     * Reads the content of the requirements file
     * @return the content of the requirements file as a String
     * @throws IOException if file reading fails
     */
    public static String readRequirements() throws IOException {
        return readRequirements(DEFAULT_INPUT_PATH);
    }
    
    /**
     * Reads the content of a requirements file from the specified path
     * @param filePath the path to the requirements file
     * @return the content of the file as a String
     * @throws IOException if file reading fails
     */
    public static String readRequirements(String filePath) throws IOException {
        Path path = Paths.get(filePath);
        
        if (!Files.exists(path)) {
            throw new IOException("Requirements file not found at: " + filePath);
        }
        
        if (!Files.isReadable(path)) {
            throw new IOException("Requirements file is not readable: " + filePath);
        }
        
        String content = Files.readString(path);
        
        if (content == null || content.trim().isEmpty()) {
            throw new IOException("Requirements file is empty: " + filePath);
        }
        
        System.out.println("Successfully read requirements from: " + filePath);
        System.out.println("Content length: " + content.length() + " characters");
        
        return content.trim();
    }
    
    /**
     * Checks if the requirements file exists at the default location
     * @return true if the file exists, false otherwise
     */
    public static boolean requirementsFileExists() {
        return Files.exists(Paths.get(DEFAULT_INPUT_PATH));
    }
}

File: nextjs/src/utils/requirementReader.ts
import { promises as fs } from "fs";
import path from "path";

const DEFAULT_INPUT_PATH = "input/requirements.txt";

/**
 * Reads the content of the requirements file at the given path or the default path.
 * Throws an Error if file is missing, unreadable, or empty.
 */
export async function readRequirements(filePath?: string): Promise<string> {
  const relative = filePath || DEFAULT_INPUT_PATH;
  const resolved = path.resolve(process.cwd(), relative);

  try {
    await fs.access(resolved, fs.constants.R_OK);
  } catch (err) {
    throw new Error(`Requirements file not found or not readable: ${relative}`);
  }

  const content = await fs.readFile(resolved, { encoding: "utf8" });
  if (!content || content.trim() === "") {
    throw new Error(`Requirements file is empty: ${relative}`);
  }

  console.log(`Successfully read requirements from: ${relative}`);
  console.log(`Content length: ${content.length} characters`);

  return content.trim();
}

/**
 * Convenience wrapper to read from the default input file
 */
export async function readDefaultRequirements(): Promise<string> {
  return readRequirements(DEFAULT_INPUT_PATH);
}

/**
 * Returns true if the default requirements file exists
 */
export async function requirementsFileExists(): Promise<boolean> {
  const resolved = path.resolve(process.cwd(), DEFAULT_INPUT_PATH);
  try {
    await fs.access(resolved, fs.constants.F_OK);
    return true;
  } catch {
    return false;
  }
}
